backend/src/layer/data/flickr_api/json_parser.rs

```diff
@@ -0,0 +1,234 @@
+use crate::interface::image_hoster::model::ImageMetaData;
+use crate::interface::image_hoster::ImageHosterError;
+use crate::layer::data::flickr_api::json_structs::*;
+
+pub struct JSONParser;
+
+fn get_selected_size_fallback() -> String { String::from("Medium") }
+fn get_selected_size() -> String { String::from("Large") }
+
+impl JSONParser {
+    pub const fn new() -> Self {
+        Self
+    }
+
+    /// Obtains the preferred image information from the [`JsonRootSizes`] struct.
+    /// # Return
+    /// The [`ImageMetaData`] struct containing all necessary information for the image.
+    /// If the preferred size cannot be obtained a fallback to a smaller size 'll be done.
+    /// If even this fallback size is not available the url will be empty. No url 'll be provided.
+    pub fn parse_get_sizes(&self, root: JsonRootSizes, photo_id: &str) -> ImageMetaData {
+        let mut url = String::new();
+        for size in root.sizes.size.clone() {
+            if size.label == get_selected_size() {
+                url = size.source;
+                break;
+            }
+        }
```

Comment:
```suggestion
        let url = root.sizes.size.iter()
            .find(|s| s.label == get_selected_size())
            .map(|s| s.source)
            .unwrap_or_default();
```
can't the world be beautiful?